classes/admin.class.php

```diff
@@ -264,31 +427,74 @@ function ppom_attach_ppoms() {
 			wp_send_json( $response );
 		}
 
-		// wp_send_json($_POST);
-		$response = array();
-
-		$ppom_id = intval( $_POST['ppom_id'] );
-
+		$ppom_id      = intval( $_POST['ppom_id'] );
 		$ppom_udpated = 0;
-		// Reset
-		if ( isset( $_POST['ppom_removed'] ) ) {
 
-			foreach ( $_POST['ppom_removed'] as $product_id ) {
+		// +----- Attach Field to Product -----+
+		$products_to_attach = isset( $_POST['ppom-attach-to-products'] ) && is_array( $_POST['ppom-attach-to-products'] ) ? array_map( 'sanitize_key', $_POST['ppom-attach-to-products'] ) : array();
 
-				delete_post_meta( intval( $product_id ), PPOM_PRODUCT_META_KEY );
-				$ppom_udpated ++;
-			}
-		}
+		$products = wc_get_products(
+			array(
+				'limit'   => -1,
+				'return'  => 'ids',
+			)
+		);
+
+		$products = array_map('intval', $products );
 
-		if ( isset( $_POST['ppom_attached'] ) ) {
+		foreach ( $products as $product_id ) {
```

Comment:
Here, the logic to loop through all the products to run those checks looks suboptimal to me, and I think it used to be better and faster. 

 I suggest we reconsider and do either of those: 
1. Save the initial product list from the frontend side in a JS variable before the user modifies it and sends it via AJAX, as initial_products. When saving, we see if there is a difference and just run delete for missing and adding/updating to the new ones. 
2. Or, we run a cleanup before the addition. I.e. we delete the metals first for all products via some custom wpdb query, and afterward, we add it to the new ones. 

I think either of those will be faster versus looping through all product catalogs, which can be very slow in stores with a lot of products.

Reply:
Thanks for the suggestion. I will go with the first approach.

The changes are like this:
- added a hidden input with the original values (No need for extra JS overload; it will be automatically sent to the server)
- The attach and delete workflow will work only on the differences between the initial and given values from select.